### PR TITLE
Update license info

### LINKED ISSUE #2827: Update license information for files under build-aux/m4
https://github.com/zcash/zcash/blob/master/build-aux/m4/l_atomic.m4#L4 says that file is sourced from http://bugs.debian.org/797228 , which in turn references this file from Filezilla: https://svn.filezilla-project.org/filezilla/FileZilla3/trunk/m4/check_atomic.m4?revision=6938&view=markup&pathrev=6938

Filezilla is [GPLv2-licensed](https://svn.filezilla-project.org/filezilla/FileZilla3/trunk/COPYING?view=markup), and I cannot find any exception for the license of the Autoconf macros it uses.

This is a problem because the GPLv2 license affects everything else. Note that the output of the macro is arguably *also* implicitly GPLv2-licensed as a derived work of the macro (since there's no exception that would exclude it from being so licensed). I think we need to replace this file, or get permission to include it under a more permissive license.

### PR DESCRIPTION
**Pull request by daira**
Includes commits cherry-picked from bitcoin/bitcoin#8784 with some minor changes.

Fixes #2827.

### CODE REVIEW

**Comment by daira** [comment]
This includes #3505, and can be rebased once that is merged.

**Comment by daira** [comment]
Rebased after merging #3505.

File: contrib/debian/copyright [hunk 1]
```diff
@@ -8,18 +8,98 @@ Copyright: 2016-2019, The Zcash developers
...
+Files: build-aux/m4/ax_boost_program_options.m4
+Copyright: 2009, Thomas Porschberg <thomas@randspringer.de>
+License: GNU-All-permissive-License
+
+Files: build-aux/m4/ax_boost_system.m4
+Copyright: 2008, Thomas Porschberg <thomas@randspringer.de>
+ 2008, Michael Tindal
+ 2008, Daniel Casimiro <dan.casimiro@gmail.com>
+License: GNU-All-permissive-License
+
```
Comment (str4d) [inline contrib/debian/copyright]: Missing entry:

```suggestion

Files: build-aux/m4/ax_boost_thread.m4
Copyright: 2009, Thomas Porschberg <thomas@randspringer.de>
 2009, Michael Tindal
License: GNU-All-permissive-License

```

**Comment by ravibhojwani** [comment]
I note some prof is claiming the math .. what a muppet .. who is he .. ?

**Comment by sandakersmann** [comment]
These links http://www.opensource.org/licenses/mit-license.php should be updated to https://

**Comment by daira** [comment]
Reminder: #4060 removes some autoconf macros.

File: contrib/debian/copyright (hunk 1 above)
Comment (daira) [reply contrib/debian/copyright]: Thanks, will fix.

**Comment by daira** [comment]
Rebased and addressed @sandakersmann 's comment.

**Comment by Eirik0** [comment]
Does this address https://github.com/zcash/zcash/issues/2887?

**Comment by daira** [comment]
No.

File: COPYING [hunk 2]
```diff
@@ -36,3 +38,16 @@ Although almost all of the Zcash code is licensed under "permissive" open source
 licenses, users and distributors should note that when built using the default
 build options, Zcash depends on Oracle Berkeley DB 6.2.x, which is licensed
 under the GNU Affero General Public License.
+
+
+Contributors should understand licensing implications before modifying the
+following files in build-aux/m4 (see ticket #2827):
```
Comment (Eirik0) [inline COPYING]:
```suggestion
following files in build-aux/m4 (see https://github.com/zcash/zcash/issues/2827):
```

**Comment by daira** [comment]
@zkbot r+

**Comment by daira** [comment]
`zcutil/fetch-params.sh --testnet` failed on some builders. @zkbot retry

**Pull request merged**